### DISCUSSION TITLE
How to use video.js instead of the default ui.video?

### DISCUSSION THREAD

**Discussion by luowyang** [zauberzeug/nicegui#497] [Q&A] [answered]
I have been working on a project to interactively display various videos in different formats. The native HTML5 video player certainly does not fulfill this purpose. So I turn to [video.js](https://videojs.com/). However, as someone new to the front-end world, I found it confusing and arcane to get video.js to work with NiceGUI. Can anyone kindly provide a working example or point me to some resources?

**Comment by rodja** [comment, marked as answer]
Thanks for reaching out. Have a look at our [custom Vue component example ](https://github.com/zauberzeug/nicegui/tree/main/examples/custom_vue_component) for inspiration. We also have some real [elements](https://github.com/zauberzeug/nicegui/blob/main/nicegui/elements/) which could be a blueprint:

- aggrid
- mermaid
- plotly
